src/CarbonAware.DataSources/CarbonAware.DataSources.ElectricityMaps/test/Client/ElectricityMapsClientTests.cs

```diff
@@ -107,7 +113,12 @@ public void AllPublicMethods_DoNotSwallowBadProxyExceptions()
     public void GetForecastedCarbonIntensityAsync_ThrowsWhenBadJsonIsReturned()
     {
         // Arrange
-        CreateBasicClient(TestData.GetZonesAllowedJsonString(), "This is bad json");
+        AddHandlers_Auth_Zones(TestData.GetZonesAllowedJsonString());
```

Comment:
Seems like this statement is used in multiple methods. Would it be possible to create test cases for this? It feels repetitive.

Reply:
Condensed these tests since they catch the same case across all the routes